### PR TITLE
Add :insert-text command

### PR DESCRIPTION
**Pull request by blyxxyz**
Adds an `:insert-text` command for inserting text and deprecates `:paste-primary` to `:insert-text {primary}`.

See #1790.

### CODE REVIEW

**Comment by The-Compiler** [comment]
@lahwaacz want to help review this since you originally implemented `:paste-primary`? :smile:

**Comment by The-Compiler** [comment]
This LGTM now - waiting a bit for @lahwaacz to speak up - and then planning to merge this later today or tomorrow.

File: tests/end2end/features/yankpaste.feature [hunk 1]
```diff
@@ -227,82 +227,59 @@ Feature: Yanking and pasting.
...
         And I press the keys "<Home>"
-        And I press the key "<Ctrl+Right>"
-        And I press the key "<Ctrl+Right>"
-        And I run :paste-primary
+        And I press the key "<Right>"
+        And I press the key "<Right>"
+        And I run :insert-text Hello world
         # Compare
-        Then the text field should contain "one two Hello world three four"
+        Then the text field should contain "onHello worlde two three four"
```
Comment (lahwaacz) [inline tests/end2end/features/yankpaste.feature]: Not that it's important, but why did you change the insert position? And the comment above still says "two words to the right" :wink:

Comment (The-Compiler) [inline tests/end2end/features/yankpaste.feature]: See the commit message of e7367bd949a7ad85327e27f4cec0858a3ec666ff (though it seemed to work fine so far on Travis and Appveyor?)

Comment (blyxxyz) [inline tests/end2end/features/yankpaste.feature]: I removed the leading space so I could use "Hello world" as a direct argument to `:insert-text`, and for some reason it ended up as `one twoHello world three four` on Linux and `one two Hello worldthree four` on Windows.

File: tests/end2end/test_insert_mode.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def test_insert_mode(file_name, source, input_text, auto_insert, quteproc):
         quteproc.press_keys(input_text)
     elif source == 'clipboard':
         quteproc.send_cmd(':debug-set-fake-clipboard "{}"'.format(input_text))
-        quteproc.send_cmd(':paste-primary')
+        quteproc.send_cmd(':insert-text {clipboard}')
```
Comment (lahwaacz) [inline tests/end2end/test_insert_mode.py]: Is this `{clipboard}` or `{primary}`, as the deprecation message for `:paste-primary` says?

**Comment by lahwaacz** [comment]
Somehow I don't see any code which would do the replacement for `:insert-text {primary}`. In any case, it should be tested.

File: tests/end2end/test_insert_mode.py (hunk 2 above)
Comment (blyxxyz) [inline tests/end2end/test_insert_mode.py]: I changed it to `{clipboard}` to make it work when the primary selection is not supported. `:paste-primary` fell back on the regular clipboard, but `{primary}` doesn't.

**Comment by The-Compiler** [comment]
@lahwaacz: @blyxxyz did add that (including some tests with `:open`) in #1791 - but I thought the same first :laughing:

**Comment by lahwaacz** [comment]
Oh, that's on the other end of my to-do list :laughing: 

Anyway, now that I partially caught up on these changes, all the replacement pseudovariables are missing in the documentation - which will make it particularly hard to find.

File: tests/end2end/features/yankpaste.feature (hunk 1 above)
Comment (lahwaacz) [inline tests/end2end/features/yankpaste.feature]: Hmm, then the previous behaviour probably relied on the leading space and Windows did not interpret that literally.

File: tests/end2end/test_insert_mode.py (hunk 2 above)
Comment (lahwaacz) [inline tests/end2end/test_insert_mode.py]: OK, I'm fine with that as long as it works :+1:

**Comment by The-Compiler** [comment]
> all the replacement pseudovariables are missing in the documentation - which will make it particularly hard to find.

I'm not sure there is a good place in the documentation to put them... I really want to write a real documentation, but that's kind of blocked by switching to Sphinx/ReST (#345).

**Comment by lahwaacz** [comment]
I think the right place for this would be `qute://help/commands.html`, between the "Commands" and "Normal commands" headers :wink:

As for the doc generation, the current `src2asciidoc.py` script already knows how to extract docstrings, so we could use a module docstring to hold the info - maybe `qutebrowser/commands/__init__.py`?

**Comment by The-Compiler** [comment]
Thanks @blyxxyz and @lahwaacz! I liked that idea, and added some basic docs in 5ecef0004cb7b7fc5f504bc6fbc48fa16e7c247c.

File: tests/end2end/features/yankpaste.feature (hunk 1 above)
Comment (The-Compiler) [inline tests/end2end/features/yankpaste.feature]: Yeah, for some reason those are platform-specific, even when [doing the same thing via the API](https://github.com/The-Compiler/qutebrowser/blob/v0.8.2/qutebrowser/browser/webkit/webkittab.py#L190-L214) we need to do some extra stuff for Windows for some reason.
